module/VuFind/src/VuFind/Sitemap/Plugin/IndexFactory.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Index-based generator plugin factory
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2021.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Sitemap
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\Sitemap\Plugin;
+
+use Interop\Container\ContainerInterface;
+use Interop\Container\Exception\ContainerException;
+use Laminas\ServiceManager\Exception\ServiceNotCreatedException;
+use Laminas\ServiceManager\Exception\ServiceNotFoundException;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+
+/**
+ * Index-based generator plugin factory
+ *
+ * @category VuFind
+ * @package  Sitemap
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class IndexFactory implements FactoryInterface
+{
+    /**
+     * Create an object
+     *
+     * @param ContainerInterface $container     Service manager
+     * @param string             $requestedName Service being created
+     * @param null|array         $options       Extra options (optional)
+     *
+     * @return object
+     *
+     * @throws ServiceNotFoundException if unable to resolve the service.
+     * @throws ServiceNotCreatedException if an exception is raised when
+     * creating a service.
+     * @throws ContainerException&\Throwable if any other error occurs
+     */
+    public function __invoke(
+        ContainerInterface $container,
+        $requestedName,
+        array $options = null
+    ) {
+        if (!empty($options)) {
+            throw new \Exception('Unexpected options passed to factory.');
+        }
+        $configLoader = $container->get(\VuFind\Config\PluginManager::class);
+        $sitemapConfig = $configLoader->get('sitemap');
```

Comment:
I'm slightly concerned that the Index plugin is not completely controlled by the generator but reads the config on its own. This makes it less controllable than I'd find ideal, but since I can't come up with a real world example of where this would be an issue, and I also don't can't come up with a really clean way of providing the necessary config from the Generator, I think it's okay.

Reply:
I thought it was actually a benefit to separate concerns here, and allow the Generator to know as little as possible about the plugin. This allows custom configuration to be injected with a custom factory, instead of having to modify the Generator... and it wouldn't be too hard to have multiple subclasses with multiple factories if there were somehow a need to have multiple instances of the Index plugin running with different configurations.

All that being said, if you do think of a reason to change the approach, I'm always open to further discussion.

Reply:
Haha, I see! And that's valid reasoning as well, so no further objections..